### PR TITLE
Update bot presence?

### PR DESCRIPTION
**Pull request by starspritechippy**
I'm not sure if this will work, but maybe, because the bot is a little funky on startup, this *could* fix this issue?
Definitely, the huge amount of servers and users the bot connects to plays into this.

Especially the part where you mentioned "Has worked always" makes me think this might be the issue.

I also thought it might have something to do with the bot not being ready for a long time. Last time I saw you restart the bot, when you checked `bot.is_ready()` repeatedly, it took about ten minutes for it to finally be ready.

Another possibility would be to remove the line 93.

Of course there's no easy way to test this, maybe it would be possible to test using the beta bot?

### CODE REVIEW

**Comment by Gelbpunkt** [comment]
have to mention that it's the `on_ready` event so I don't see why we should wait for it to be ready...

**Comment by numbermaniac** [comment]
Perhaps instead of using `change_presence` after the bot has been logged in, could we use the [activity keyword](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.activity) of commands.Bot directly?

**Comment by starspritechippy** [comment]
I put in some new commits so it includes Number's idea.
A different idea to approach this would be to create a background task that cycles through multiple messages. That way you could have more messages in the bot's presence but this would probably also guarantee that it has a presence in the first place.
Let me know what you guys think and if that makes sense.

File: idlerpg.py [hunk 1]
```diff
@@ -31,6 +31,11 @@
 
 bot = Bot(
     case_insensitive=True,
+    activity=activity=discord.Game(
```
Comment (numbermaniac) [inline idlerpg.py]: `activity=activity=` I think you only want one of these. :)

Comment (starspritechippy) [reply idlerpg.py]: I should probably wake up before going on GitHub 😅 
Good thing you noticed

**Comment by Gelbpunkt** [comment]
I'd rather do it in the bot subclass' `__init__` to keep it modular

**Comment by starspritechippy** [comment]
hope that does the trick

**Comment by Gelbpunkt** [comment]
yea should do, would you mind running the format.sh script over your pull requests to assure the syntax looks as intended? :)

**Comment by starspritechippy** [comment]
image
Still giving some errors, but it doesn't seem like anything was changed there?
Might just be that black doesn't yet know about `:=`.

**Comment by Gelbpunkt** [comment]
if you run `pip3 install -U -r requirements-dev.txt`, it'll install python 3.8 syntax compatible versions

**Comment by starspritechippy** [comment]
Done that, seems like it didn't change anything. I even got one *more* error.

`.\bot.py:461:66: E225 missing whitespace around operator`

But looking at it overall, and how the rest of the bot is formatted, this looks correct to me.

**Comment by Gelbpunkt** [comment]
weird, maybe pip didn't update them as they're already installed. It worked fine for me in a clean install without black, flake8 and isort. Maybe try uninstalling them and rerun my command?

**Comment by starspritechippy** [comment]
`pip3 uninstall`ed those three packages, reinstalled them using your command, but it still gives the same error. Maybe, just maybe, it might be because I use Git Bash on Windows? Maybe it works differently on Linux? I'm not sure.

**Comment by Gelbpunkt** [comment]
try uninstalling pyflakes before, that's what flake8 uses for syntax matching

**Comment by starspritechippy** [comment]
Same result again :(

**Comment by Gelbpunkt** [comment]
well I'll run it then when I've merge the PRs. Too hyped already :)

**Comment by starspritechippy** [comment]
Let's hope it actually works, that's still not guaranteed :p

**Comment by numbermaniac** [comment]
It looks like the `status` parameter can be used like `commands.Bot(status = discord.Status.idle)` to also pass in the Idle status when creating the bot.

https://github.com/Rapptz/discord.py/blob/master/discord/state.py#L89-L94

**Comment by starspritechippy** [comment]
@numbermaniac should I also put that in the `__init__` function?

**Comment by numbermaniac** [comment]
If you do it in `__init__` instead of where the `commands.Bot` instance is created, I think you'd have to do:
```py
self._connection._status = "idle"
```

**Comment by starspritechippy** [comment]
I'll stick with the way I'm used to.

**Pull request merged**